src/RouterSelector.web.js

```diff
@@ -97,18 +149,35 @@ let AppRouter = React.lazy(() => {
 })
 
 const RouterSelector = () => {
+  const [showDialog] = useDialog()
   const store = SimpleStore.useStore()
+  const [displayUpgradeDialog, setDisplayUpgradeDialog] = useState(false)
 
   //we use global state for signup process to signal user has registered
   const isLoggedIn = store.get('isLoggedIn') //Promise.resolve( || AsyncStorage.getItem(IS_LOGGED_IN))
+  const serviceWorkerUpdated = store.get('serviceWorkerUpdated')
+
+  useEffect(() => {
+    log.info('service worker updated', {
+      serviceWorkerUpdated,
+      displayUpgradeDialog,
+    })
+
+    if (serviceWorkerUpdated && displayUpgradeDialog) {
+      showUpgradeDialog(showDialog, serviceWorkerUpdated)
+    }
+  }, [serviceWorkerUpdated, displayUpgradeDialog])
+
+  // upgrade dialog should be shown after the router is loaded - so wait for splash will be unmounted
+  const allowDisplayingUpgradeDialog = useCallback(() => setDisplayUpgradeDialog(true), [setDisplayUpgradeDialog])
 
   const Router = useMemo(() => {
     log.debug('RouterSelector Rendered', { isLoggedIn })
     return isLoggedIn ? AppRouter : SignupRouter
   }, [isLoggedIn])
 
   return (
-    <React.Suspense fallback={<Splash animation />}>
+    <React.Suspense fallback={<Splash animation onUnmount={allowDisplayingUpgradeDialog} />}>
```

Comment:
this looks like some king of hack.
the React-way to solve the issue that dialog could be shown once after app preloaded is **composition**
We'll just create an component which will render the proper router depneding of isLoggedIn prop passed
and will execute hook that checks updated flag in the store

Please:

1. create an separate component NestedRouter (in this file)

```javascript
const NestedRouter = ({ isLoggedIn, ...props }) => {
   const RouterComponent = useMemo(() => {
      log.debug('RouterSelector Rendered', { isLoggedIn })
      return isLoggedIn ? AppRouter : SignupRouter
    }, [isLoggedIn])

    return <RouterComponent {...props} />
}
```

if you don't need additional props and they won't be needed in the near future, the component could be most simplier:

```javascript
const NestedRouter = ({ isLoggedIn }) => {
   //  logging isLoggedIn changes
   useEffect(() => {
      log.debug('RouterSelector Rendered', { isLoggedIn })
    }, [isLoggedIn])

    return isLoggedIn ? <AppRouter /> : <SignupRouter />
}
```

2. add the  hook from line 260 inside, but without displayUpgradeDialog condition
3. add also everything this hook required (e.g. showDialog, store, serviceWorkerUpdated
4. remove displayUpgradeDialog state var
5. replace <Router /> to <NestedRouter isLoggedIn={isLoggedIn} />